lib/logstash/inputs/jdbc.rb

```diff
@@ -232,4 +246,30 @@ def update_state_file
     end
   end
 
+  private
+
+  def enable_encoding
+    !@default_encoding.nil? || (@default_encoding.nil? && !@columns_to_encode.empty?)
+  end
+
+  # make sure the encoding is uniform over fields
+  def convert(key, value)
+    return value unless should_convert(key, value)
+    converter = find_converter_for(key, value)
+    converter.convert(value)
+  end
+
+  def should_convert(key, value)
+    column_keys = @columns_to_encode.keys
+    value.is_a?(String) && (!@default_encoding.nil? || column_keys.include?(key))
+  end
+
+  def find_converter_for(key, value)
+    encoding = @columns_to_encode[key] || @default_encoding
+    unless @converters.keys.include?(encoding)
+      @converters[encoding] = LogStash::Util::Charset.new(encoding)
+    end
+    @converters[encoding]
+  end
```

Comment:
This method also feels inefficient? Unless I am missing something, the same query should yield the same columns so this `@converters` initialization could be all done on the very first row and then just do a simple lookup in `@converters` for the rest no?